apis/v1alpha2/gatewayclass_types.go

```diff
@@ -47,12 +60,19 @@ type GatewayClass struct {
 	Status GatewayClassStatus `json:"status,omitempty"`
 }
 
+const (
+	// GatewayClassFinalizerGatewaysExist MUST be added as a finalizer to the
+	// GatewayClass whenever there are provisioned Gateways using a
```

Comment:
is it really a MUST? Does IngressClass have this?

IMO every product that uses `finalizers` has a bad user experience. I am fairly strongly against using one in our product. There is nothing worse than things not deleting when you want them too...

Reply:
I don't think we _need_ to enforce this behavior, there is no parallel for IngressClass. One thing that we need to be clear about though it should not be easy to delete all Gateways of a certain class simply by deleting the GatewayClass.

Reply:
Moved back to "should" here to match previous state.